### PR TITLE
boot: convert paging from assembly to Zig (take 2)

### PR DESCRIPTION
**Pull request by mewmew**
The main focus of this PR is to make it easier to work with paging.

For this reason, the kernel code and data segments are treated separately by dedicated functions, and each function has a start index (e.g. `P2_KERNEL_STACK_FIRST_INDEX`) to make it easier to rearrange the relative order of page mappings in the future.

This PR supersedes #19.

### CODE REVIEW

**Comment by mewmew** [comment]
Zig 0.12 lyckades åstadkomma ett stenras i Lappis, hehe, boot-asm rewrite'n funkar inte längre.

**Comment by mewmew** [comment]
Userland is now working with the following paging table.
In particular, user accessibility has now been removed from the kernel code/data and stack segments.
```
gef➤  pt -phys_verbose
             Address :         Length :         Phys | Permissions          
                 0x0 :      0x4000000 :          0x0 | W:1 X:1 S:1 UC:0 WB:1
           0x4000000 :       0x200000 :    0x4000000 | W:0 X:0 S:1 UC:0 WB:1
           0x4200000 :       0x400000 :    0x4200000 | W:1 X:0 S:1 UC:0 WB:1
           0x4600000 :       0x200000 :    0x4600000 | W:0 X:0 S:1 UC:0 WB:1
           0x4800000 :       0x200000 :    0x4800000 | W:0 X:1 S:0 UC:0 WB:1
           0x4a00000 :      0xfc00000 :    0x4a00000 | W:1 X:1 S:0 UC:0 WB:1
          0x14600000 :       0x200000 :   0x14600000 | W:0 X:1 S:0 UC:0 WB:1
          0x14800000 :       0x600000 :   0xfd000000 | W:1 X:0 S:1 UC:0 WB:1
```

**Comment by mewmew** [comment]
CI build requires `lld`. From https://github.com/karlek/lappis/actions/runs/8858873900/job/24328021167?pr=23#step:9:41:
> make: ld.lld: No such file or directory

**Comment by karlek** [comment]
I'll fix the CI build and then merge after a short review.

File: debug.sh [hunk 1]
```diff
@@ -22,7 +22,7 @@ kitty --class qemu-starter \
 	-drive media=disk,index=0,file=bin/zipfs.img,format=raw,if=ide \
 	-cdrom bin/kernel.iso &
 
-gdb \
+kitty --class lappis-serial-raw -e gdb \
```
Comment (mewmew) [inline debug.sh:25]: Fet quality of life!

**Comment by karlek** [comment]
I'll merge this now, huge success!

**Pull request merged**